extend ExplicitAttrs to use JSON/JEXL (#1180)
telefonicaid/iotagent-node-lib: doc/api.md

| `static_attributes`            | `staticAttributes`             | this attributes will be added to all the entities of this group 'as is', additional `metadata` is optional.                                                                                                                                                               |
| `internal_attributes`          | `internalAttributes`           | optional section with free format, to allow specific IoT Agents to store information along with the devices in the Device Registry.                                                                                                                                       |
| `expressionLanguage`           | `expresionLanguage`            | optional boolean value, to set expression language used to compute expressions, possible values are: legacy or jexl. When not set or wrongly set, `legacy` is used as default value.                                                                                      |

> | `explicitAttrs`                | `explicitAttrs`                | optional boolean value, to support selective ignore of measures so that IOTA doesn’t progress. If not specified default is `false`. Additionally `explicitAttrs` can be used to define which meassures defined in JSON/JEXL array will be propagated to NGSI interface.   |

Reviewer: I suggest to add a link to advanced-topics.md#explicitly-defined-attributes-explicitattrs at this point, for a more detailed explanation of explicitAttrs functionality.

Reviewer: In addition, I understand that the explicitAttrs entry at the device API table should be also adjusted in a similar way.

Author: done in https://github.com/telefonicaid/iotagent-node-lib/pull/1180/commits/cd1fe6af5fdad079b750971b365646ec4ecbcd8b

> | `explicitAttrs`                | `explicitAttrs`                | optional boolean value, to support selective ignore of measures so that IOTA doesn’t progress. If not specified default is `false`. Additionally `explicitAttrs` can be used to define which meassures defined in JSON/JEXL array will be propagated to NGSI interface. [See all cases](advanced-topics.md#explicitly-defined-attributes-explicitattrs)  |

Reviewer: Fixed in 77436cfa00748305dfeb8feb320acc89890efe9b

| `internal_attributes` | `internalAttributes` | List of internal attributes with free format for specific IoT Agent configuration                                                                                                                                                                                         | LWM2M mappings from object URIs to attributes |
| `static_attributes`   | `staticAttributes`   | List of static attributes to append to the entity. All the updateContext requests to the CB will have this set of attributes appended.                                                                                                                                    | `[ { "name": "attr_name", "type": "Text" } ]` |
| `expressionLanguage`  | `expresionLanguage`  | optional boolean value, to set expression language used to compute expressions, possible values are: legacy or jexl. When not set or wrongly set, legacy is used as default value.                                                                                        |

> | `explicitAttrs`       | `explicitAttrs`      | Optional boolean value to support selective ignore of measures for device so that IOTA doesn’t progress. If not specified default is `false`. Additionally `explicitAttrs` can be used to define which meassures defined in JSON/JEXL array will be propagated to NGSI interface. [See all cases](advanced-topics.md#explicitly-defined-attributes-explicitattrs)                                                                            | `true/false`                                  |

Reviewer: I think is better to describe it in a simplified way and provide the detailed information in the main section:

```suggestion
| `explicitAttrs`       | `explicitAttrs`      | optional field to support selective ignore of measures so that IOTA doesn’t progress. See details in [specific section](advanced-topics.md#explicitly-defined-attributes-explicitattrs)                                                                             | (see details in specific section)                                  |
```

Reviewer: Fixed in 8a3b2ed949a7dac7be09931bcc000ac4f1bca90f
